Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift

```diff
@@ -212,6 +212,14 @@ open class AVFoundationPlayback: Playback {
     open override func play() {
         if player == nil {
             setupPlayer()
+
+            let queue = DispatchQueue(label: "audioSelectionQueue", qos: .utility)
+            queue.async {
+                self.selectDefaultAudioIfNeeded()
```

Comment:
Some questions regarding moving the code to a different async queue:
1. Does all the code from `selectDefaultAudioIfNeeded` need to be moved a different queue or just part of it ?
2. The sequence of events will be changed: `didFindAudio` will be triggered after `willPlay`. What is the impact of this change on existing plugins ?
3. How much time does it take before the audio is actually selected ? Can it happen after the video is already playing (after `playing`) ?
4. What would happen if the `AVFoundationPlayback` is destroyed before the code is finished ? Can we have a memory leak or two simultaneous `AVPlayer`'s running due to hard linking ?

Reply:
1 - There is a part of the code that is needed to be moved, but it's a part that we don't have access to make changes.

2 - It will not impact because the events still remain consistent with the state machine described in the documentation.

3 - The original operation is taking about 1 second. In our test (including a poor internet connection) we didn't get this scenario. The audio change has always respected our rules.

4 - No, because DispatchQueue is not a retain cycle. When is destroyed the references will be clean.

Reply:
1. Ok. Moving only part of the code to the async queue would increase the solution complexity without any significant gain.

2. Ok. The main issue is plugin impact due to event order changes (as it did happen in this PR). No existing plugin is affected by this change.

3. There is still a risk of changing the audio after the `playing` event. On the other hand, it was not reproduced in our tests. I suggest adding an error log to detect this situation.

4. Ok. It is not true that the references are destroyed automatically by the DispatchQueue. Latest commit, that replaces hard by weak linking, actually solves the issue.